#!/usr/bin/env python
PACKAGE = "dynamic_reconfigure"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#gen.add(parameter name, parameter type, bit mask, parameter description, default value, minimum value, maximum value)
gen.add( "max_angular_velocity", double_t, 0, "max_angular_velocity", 0.5, 0, 10)
gen.add( "max_linear_velocity", double_t, 0, "max_linear_velocity", 0.5, 0, 10)
gen.add( "Ki", double_t, 0, "Propotional Gain", 0.01, 0, 5)
gen.add( "Kp", double_t, 0, "integral gain", 1.5, 0, 10)
gen.add( "safetyDistance", double_t, 0, "Safety Distance", 2, 0, 5)

#gen.generate(namespace's name ,node's name, current file's name)
exit(gen.generate(PACKAGE, "laser_tutorial", "Laser_Params"))
